### PR TITLE
Task indicator

### PR DESCRIPTION
**Pull request by VC103221**
## Notable Changes
- Added `@twilio/flex-ui` dependency
- Added `FlexV1-Dev-Env.md` as abbreviated guide to running the v1 plugin in local dev environment
- Added `TaskIndicator` component to flag pending Tasks in the Flex UI depending on which TaskQueue the Task is in

### Setup
- Set your instance of the Twilio CLI to use a Staging account
- Clone the repo
- Checkout the `task-indicator` branch
- Run `npm ci` and copy and paste the Staging account Token (this will make sure the repo is set up with the correct environment variables)
- Run `npm run start:local:v1`
- When you get to the Flex UI login screen, click the "Login with Twilio" link at the bottom of the page
image
- Make sure you have the following skills applied to your account:
  - `testing`
  - `Calls`
  - `Chat/SMS`
  - `Spanish`
  - `tya`
  - `voicemail`
- Add your phone number to the `testBlock` array in `studio-functions/frequent-callers` from the Twilio Console > Functions and Assets > Services
- Save the change to `studio-functions/frequent-callers` and deploy

### Calls
- Call `+1 202 873 8910` once for each combination of prompts:
  - Press `2` for Spanish and do not press anything for the placeholder TYA prompt
  - Press `2` for Spanish and press `3` for the placeholder TYA prompt
  - Do not press anything for the Spanish prompt and press `3` for the placeholder TYA prompt
  - Do not press anything for the Spanish prompt and do not press anything for the placeholder TYA prompt
- Each of these test calls should generate a new Task which will pop up on the Flex UI page. You should see indicators reflecting the choices you made in the IVR portion of the call. Currently there are only two indicators possible: `TYA Help-Seeker` and `Spanish Help-Seeker`. If you did not press anything during the IVR portion of the call, the indicator section will not appear for the Task.
image
image
image
image

### Texts/SMS
- Text `+1 202 873 8910` once for each combination of prompts: (if the HelpLine is closed at the time of testing, you can start by sending `VC103221` to bypass the HelpLine status check and go straight to the Studio Flow)
  - Do not respond with `TYA` when prompted (you can respond with anything else other than `TYA` or `tya` to speed up the Studio Flow - otherwise it will wait for about 3 minutes for you to answer)
  - Respond with `TYA` when prompted
- In order to complete this portion of testing, you will need to wait at least 3 minutes before sending your first text for the first Session and the first text for the second Session. The Twilio Proxy Sessions on the Staging account are set to timeout after 3 minutes. 
- Similar with the calls, each text session should generate a new Task which will pop up on the Flex UI page with the `TYA Help-Seeker` indicator when appropriate.
image
image

### Webchat
- Run a local instance of the webchat and submit two Sessions, one for each combination of prompts: (if the HelpLine is closed at the time of testing, you can start by sending `VC103221` to bypass the HelpLine status check and go straight to the Studio Flow)
  - Do not respond with `TYA` when prompted (you can respond with anything else other than `TYA` or `tya` to speed up the Studio Flow - otherwise it will wait for about 3 minutes for you to answer)
  - Respond with `TYA` when prompted
- Each chat session should generate a new Task which will pop up in the Flex UI page with the `TYA Help-Seeker` indicator when appropriate.
image
image

### Voicemail
- Remove your `tya` skill and call `+1 202 873 8910` once for each combination of prompts and leave a test voicemail for each call:
  - Press `2` for Spanish and press `3` for the placeholder TYA prompt
  - Do not press anything for the Spanish prompt and press `3` for the placeholder TYA prompt
- Re-add your `tya` skill in between each call. There should be a voicemail Task with the appropriate indicator(s).
image
image
- Remove your `Spanish` skill and call `+1 202 873 8910`. Press `2` for Spanish when prompted and leave a test message.
- Re-add your `Spanish` skill. There should be a voicemail Task with the appropriate indicator.
image

### CODE REVIEW

File: FlexV1-Dev-Env.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+- [Install the Twilio CLI] (https://www.twilio.com/docs/twilio-cli/getting-started/install)
```
Comment (kcamach1) [inline FlexV1-Dev-Env.md:1]: is there an extra space in here

Comment (VC103221) [reply FlexV1-Dev-Env.md]: @kcamach1 nvm I figured it out lol yeah there is

**Comment by kcamach1** [comment]
For some reason Twilio is not letting me complete a call to the staging number. Error log says
> 81002 - Error event while processing widget." 

[Example](https://console.twilio.com/us1/monitor/logs/debugger/errors/NO99aa2af00f2ba8d21dd40c02abfcec19)

**Comment by VC103221** [comment]
> For some reason Twilio is not letting me complete a call to the staging number. Error log says
> 
> > 81002 - Error event while processing widget
> 
> [Example](https://console.twilio.com/us1/monitor/logs/debugger/errors/NO99aa2af00f2ba8d21dd40c02abfcec19)

this is because I had concurrent calls turned off for the studio flow and you technically had a call session already open (turned it off bc didn't know charging model/if overcharging was a problem). turned on concurrent calls.

**Review by kcamach1 — COMMENTED** [review]
Screenshot 2023-08-02 at 16 57 16

I tried to select Spanish TYA, but it only showed Spanish in the label. Also, some of the Spanish messages sounded like they were repeating multiple times. Not sure if that's an old problem that I implemented years ago unrelated to this update, though.

**Comment by VC103221** [comment]
@kcamach1 the only thing I can thing of is perhaps did you try to press `3` before the computer stopped talking completely? because Twilio has this great thing where if you try to input before the computer stops talking it won't register your input. I tested this morning and it seems to be working on my end.

also I didn't hear any of the Spanish audio clips repeating. for clarification, there are four that play right now for the route you are trying to test:
1. asking you if you would like to press `2` for Spanish
2. Spanish secondary greeting
3. Spanish notice that you should call `988` for crisis
4. More Spanish?

here's a screenshot of what the Studio Flow contains for reference (I circled the Say/Play blocks that you should be hearing Spanish from. I've listened through enough times that I just know when one block is playing vs another but I'm sorry I can't follow it well enough to be able to tell you what to listen for to also check for that. that being said there doesn't seem to be any routing which would prompt a re-run? and all the Say/Play blocks are set to run only once...
image

**Comment by kcamach1** [comment]
Screenshot 2023-08-04 at 12 43 50

I got it to work!


I noticed that you've set up the TYA input as a `Say/Play` widget giving the prompt, followed by a `Get Input on Call` widget to listen for the dial tones. You can actually have these be one widget, by making a `Get Input on Call` widget and adding the text to say in `say or play message` section. 
Screenshot 2023-08-04 at 12 46 36

This will also make it so that you can press 3 before the audio is over an Twilio will still record the input.

**Review by kcamach1 — COMMENTED** [review]
I keep putting my reviews as comments by mistake 🤦

**Comment by VC103221** [comment]
@kcamach1 oh u right actually that particular one was that way bc of the language difference but I reconfigured the studio flow to use two get input widgets and then set the variable based on the language chosen previously
image

**Review by kcamach1 — APPROVED** [review]
looks great!

**Pull request merged**